src/extensions/agent_modules/cloud_communication/eg_mqtt_broker_client/src/adu_mqtt_client_module.c

```diff
@@ -380,14 +346,36 @@ static int ADUC_MQTT_Client_Module_Initialize_DoWork(ADUC_MQTT_CLIENT_MODULE_STA
         // Ensure that we have hostName.
         if (IsNullOrEmpty(moduleState->mqttSettings.hostname))
         {
-            Log_Info("MQTT broker hostname is empty. Retry in %d seconds", DEFAULT_OPERATION_INTERVAL_SECONDS);
-            moduleState->initializeState = ADU_MQTT_CLIENT_MODULE_INITIALIZE_STATE_PARTIAL;
-            // TODO (nox-msft) : use retry_utils
-            moduleState->nextOperationTime = ADUC_GetTimeSinceEpochInSeconds() + DEFAULT_OPERATION_INTERVAL_SECONDS;
+            const char* hostNameFromDPS = ADUC_StateStore_GetMQTTBrokerHostname();
+            if (!IsNullOrEmpty(hostNameFromDPS))
+            {
+                Log_Info("Applying MQTT broker endpoint from DPS.'%s'", hostNameFromDPS);
+                moduleState->mqttSettings.hostname = strdup(hostNameFromDPS);
```

Comment:
It will goto done on line 367 if strdup here, but shouldn't it not goto done and continue on to get externalDeviceId on line 371?